src/main/scala/it/bitrock/dvs/api/core/poller/KafkaPoller.scala

```diff
@@ -14,17 +13,14 @@ trait KafkaPoller extends Actor with LazyLogging {
 
   val kafkaConsumerWrapper: KafkaConsumerWrapper
 
-  def schedulePoll(): Unit =
-    context.system.scheduleOnce(kafkaConfig.consumer.pollInterval)(kafkaConsumerWrapper.pollMessages())
-
-  override def preStart(): Unit = {
-    super.preStart()
-    logger.debug("Starting kafka message processor")
-    kafkaConsumerWrapper.pollMessages()
-  }
+  private val scheduledPoll: Cancellable =
```

Comment:
that's a hidden side effect executed at instantiation time. Could we move it to the prestart and let the implementer override it if necessary?

Reply:
You need a var to do this, if we agree to use it I can change it. That's the part I'm more uncertain about

Reply:
maybe a lazy val may work, let me check